src/commands/VerifyCommand.ts

```diff
@@ -0,0 +1,101 @@
+import { Message, MessageEmbed, TextChannel } from 'discord.js';
+import PrefixCommand from './PrefixCommand';
+import BotConfig from '../BotConfig';
+import Command from './Command';
+import DiscordUtil from '../util/DiscordUtil';
+import TaskScheduler from '../tasks/TaskScheduler';
+import RemovePendingVerificationTask from '../tasks/RemovePendingVerificationTask';
+
+export default class VerifyCommand extends PrefixCommand {
+	public readonly aliases = ['verify'];
+
+	public async run( message: Message, args: string ): Promise<boolean> {
+		if ( args.length ) {
+			return false;
+		}
+
+		const pendingChannel = await DiscordUtil.getChannel( BotConfig.verification.pendingVerificationChannel );
+
+		if ( pendingChannel instanceof TextChannel ) {
+
+			let foundUser = false;
+
+			const allMessages = pendingChannel.messages.cache;
+
+			allMessages.forEach( async thisMessage => {
+				if ( thisMessage.embeds === undefined ) return undefined;
+				if ( thisMessage.embeds[0].fields[0].value.replace( /[<>@!]/g, '' ) == message.author.id ) {
+					foundUser = true;
+				}
+			} );
+
+			try {
+				const role = await pendingChannel.guild.roles.fetch( BotConfig.verification.verifiedRole );
+				const targetUser = await message.guild.members.fetch( message.author.id );
+
+				if ( targetUser.roles.cache.has( role.id ) ) {
+					await message.channel.send( `${ message.author }, your account has already been verified!` );
+					await message.react( '❌' );
+					return false;
+				}
+			} catch ( error ) {
+				Command.logger.error( error );
+			}
+
+			if ( foundUser ) {
+				await message.channel.send( `${ message.author }, you already have a pending verification request!` );
+				await message.react( '❌' );
+				return false;
+			}
+
+		}
+
+		try {
+			const token = this.randomString( 15, '23456789abcdeghjkmnpqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ' );
```

Comment:
Token should be something like MOJIRA-DISCORD-VERIFICATION-<random>

Just to make sure people dont trick users into doing it